### DISCUSSION TITLE
nctoolkit isn't compatible with cdo 2.0.0 or above

### DISCUSSION THREAD

**Discussion by shiv3679** [pmlmodelling/nctoolkit#47] [General]
Is there any workaround where I can use nctoolkit with cdo version 2.0.4. 
![image](https://user-images.githubusercontent.com/66754219/219438728-d223f3a4-efc1-4e0e-b479-e31035873a2a.png)

**Comment by robertjwilson** [comment]
A workaround would be to clone the github repository and then make a tweak to stop nctoolkit from checking the CDO version.

So

    git clone https://github.com/pmlmodelling/nctoolkit.git

Then

    cd nctoolkit

You will then need to open nctoolkit/utils.py

Then delete the line with validate_version()

That's the line that checks which version of CDO is on the system, and it will throw an error if it's 2.0.4.

Then

    pip install .

to install from source.

nctoolkit will work fine for CDO versions 2.05 and above. It currently throws an error because of a couple of bugs in CDO 2.0.4 that were causing knock on bugs in nctoolkit.

**Reply by robertjwilson** [reply]
The error message here is also in need of an update. I've just modified the dev version so it so that it says nctoolkit is not compatible with CDO 2.0.0 to 2.0.4 and you  need to upgrade CDO.

**Reply by Gyuknj** [reply]
> A workaround would be to clone the github repository and then make a tweak to stop nctoolkit from checking the CDO version.
> 
> So
> 
> ```
> git clone https://github.com/pmlmodelling/nctoolkit.git
> ```
> 
> Then
> 
> ```
> cd nctoolkit
> ```
> 
> You will then need to open nctoolkit/utils.py
> 
> Then delete the line with validate_version()
> 
> That's the line that checks which version of CDO is on the system, and it will throw an error if it's 2.0.4.
> 
> Then
> 
> ```
> pip install .
> ```
> 
> to install from source.
> 
> nctoolkit will work fine for CDO versions 2.05 and above. It currently throws an error because of a couple of bugs in CDO 2.0.4 that were causing knock on bugs in nctoolkit.

I have been trying to install nctoolkit using this method for the past two days now but no success.

I really need urgent assistance to help me with the installations to enable me continue my analysis.

Warm regards

Gyuk Nehemiah John

**Reply by robertjwilson** [reply]
You will need to specify what is causing the problem. Does my solution not work?

**Reply by Gyuknj** [reply]
I followed the solution you suggested but it does not work. 
This is the error message I got: 
[Reg_Cor - Jupyter Notebook.pdf](https://github.com/pmlmodelling/nctoolkit/files/10949226/Reg_Cor.-.Jupyter.Notebook.pdf)

**Reply by robertjwilson** [reply]
I should have typed ‘nctoolkit/__init__.py’ in my original reply. Open that in the dev version and remove validate version and then reinstall

**Reply by Gyuknj** [reply]
Installed!

I have done the correlation with easy, thanks to your script on Github! 

How can I do the regression?

**Reply by robertjwilson** [reply]
Good to see this issue resolved. I will close it. If you have another question please raise it separately.
